// @elastic/react-search-ui-views/lib/layouts/Layout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _LayoutSidebar = _interopRequireDefault(require("./LayoutSidebar"));

var _viewHelpers = require("../view-helpers");

function Layout(_ref) {
  var className = _ref.className,
      children = _ref.children,
      header = _ref.header,
      bodyContent = _ref.bodyContent,
      bodyFooter = _ref.bodyFooter,
      bodyHeader = _ref.bodyHeader,
      sideContent = _ref.sideContent;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _viewHelpers.appendClassName)("sui-layout", className)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-header"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-header__inner"
  }, header)), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-body"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-body__inner"
  }, /*#__PURE__*/_react.default.createElement(_LayoutSidebar.default, {
    className: "sui-layout-sidebar"
  }, sideContent), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-main"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-main-header"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-main-header__inner"
  }, bodyHeader)), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-main-body"
  }, children || bodyContent), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-layout-main-footer"
  }, bodyFooter)))));
}

Layout.propTypes = {
  className: _propTypes.default.string,
  children: _propTypes.default.node,
  header: _propTypes.default.node,
  bodyContent: _propTypes.default.node,
  bodyFooter: _propTypes.default.node,
  bodyHeader: _propTypes.default.node,
  sideContent: _propTypes.default.node
};
var _default = Layout;
exports.default = _default;
