// @elastic/react-search-ui-views/lib/BooleanFacet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _types = require("./types");

var _viewHelpers = require("./view-helpers");

function BooleanFacet(_ref) {
  var className = _ref.className,
      label = _ref.label,
      options = _ref.options,
      onChange = _ref.onChange,
      onRemove = _ref.onRemove,
      values = _ref.values;
  var trueOptions = options.find(function (option) {
    return option.value === "true";
  });
  if (!trueOptions) return null;
  var isSelected = values.includes("true");

  var apply = function apply() {
    return onChange("true");
  };

  var remove = function remove() {
    return onRemove("true");
  };

  var toggle = function toggle() {
    isSelected ? remove() : apply();
  };

  return /*#__PURE__*/_react.default.createElement("fieldset", {
    className: (0, _viewHelpers.appendClassName)("sui-facet", className)
  }, /*#__PURE__*/_react.default.createElement("legend", {
    className: "sui-facet__title"
  }, label), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-boolean-facet"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-boolean-facet__option-input-wrapper"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "sui-boolean-facet__option-label"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-boolean-facet__option-input-wrapper"
  }, /*#__PURE__*/_react.default.createElement("input", {
    className: "sui-boolean-facet__checkbox",
    type: "checkbox",
    onChange: toggle,
    checked: isSelected
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sui-boolean-facet__input-text"
  }, label)), /*#__PURE__*/_react.default.createElement("span", {
    className: "sui-boolean-facet__option-count"
  }, trueOptions.count)))));
}

BooleanFacet.propTypes = {
  className: _propTypes.default.string,
  label: _propTypes.default.string.isRequired,
  onRemove: _propTypes.default.func.isRequired,
  options: _propTypes.default.arrayOf(_types.FacetValue).isRequired,
  values: _propTypes.default.arrayOf(_types.FilterValue).isRequired,
  onChange: _propTypes.default.func.isRequired
};
var _default = BooleanFacet;
exports.default = _default;
