// @elastic/react-search-ui-views/lib/MultiCheckboxFacet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _types = require("./types");

var _viewHelpers = require("./view-helpers");

function MultiCheckboxFacet(_ref) {
  var className = _ref.className,
      label = _ref.label,
      onMoreClick = _ref.onMoreClick,
      onRemove = _ref.onRemove,
      onSelect = _ref.onSelect,
      options = _ref.options,
      showMore = _ref.showMore,
      showSearch = _ref.showSearch,
      onSearch = _ref.onSearch,
      searchPlaceholder = _ref.searchPlaceholder;
  return /*#__PURE__*/_react.default.createElement("fieldset", {
    className: (0, _viewHelpers.appendClassName)("sui-facet", className)
  }, /*#__PURE__*/_react.default.createElement("legend", {
    className: "sui-facet__title"
  }, label), showSearch && /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-facet-search"
  }, /*#__PURE__*/_react.default.createElement("input", {
    className: "sui-facet-search__text-input",
    type: "search",
    placeholder: searchPlaceholder || "Search",
    onChange: function onChange(e) {
      onSearch(e.target.value);
    }
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-multi-checkbox-facet"
  }, options.length < 1 && /*#__PURE__*/_react.default.createElement("div", null, "No matching options"), options.map(function (option) {
    var checked = option.selected;
    return /*#__PURE__*/_react.default.createElement("label", {
      key: "".concat((0, _viewHelpers.getFilterValueDisplay)(option.value)),
      htmlFor: "example_facet_".concat(label).concat((0, _viewHelpers.getFilterValueDisplay)(option.value)),
      className: "sui-multi-checkbox-facet__option-label"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "sui-multi-checkbox-facet__option-input-wrapper"
    }, /*#__PURE__*/_react.default.createElement("input", {
      id: "example_facet_".concat(label).concat((0, _viewHelpers.getFilterValueDisplay)(option.value)),
      type: "checkbox",
      className: "sui-multi-checkbox-facet__checkbox",
      checked: checked,
      onChange: function onChange() {
        return checked ? onRemove(option.value) : onSelect(option.value);
      }
    }), /*#__PURE__*/_react.default.createElement("span", {
      className: "sui-multi-checkbox-facet__input-text"
    }, (0, _viewHelpers.getFilterValueDisplay)(option.value))), /*#__PURE__*/_react.default.createElement("span", {
      className: "sui-multi-checkbox-facet__option-count"
    }, option.count.toLocaleString("en")));
  })), showMore && /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: "sui-facet-view-more",
    onClick: onMoreClick,
    "aria-label": "Show more options"
  }, "+ More"));
}

MultiCheckboxFacet.propTypes = {
  label: _propTypes.default.string.isRequired,
  onMoreClick: _propTypes.default.func.isRequired,
  onRemove: _propTypes.default.func.isRequired,
  onSelect: _propTypes.default.func.isRequired,
  onSearch: _propTypes.default.func.isRequired,
  options: _propTypes.default.arrayOf(_types.FacetValue).isRequired,
  showMore: _propTypes.default.bool.isRequired,
  className: _propTypes.default.string,
  showSearch: _propTypes.default.bool,
  searchPlaceholder: _propTypes.default.string
};
var _default = MultiCheckboxFacet;
exports.default = _default;
