// @elastic/react-search-ui-views/lib/SingleLinksFacet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _types = require("./types");

var _viewHelpers = require("./view-helpers");

function SingleLinksFacet(_ref) {
  var className = _ref.className,
      label = _ref.label,
      onRemove = _ref.onRemove,
      onSelect = _ref.onSelect,
      options = _ref.options;
  var value = options.filter(function (o) {
    return o.selected;
  }).map(function (o) {
    return o.value;
  })[0];
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _viewHelpers.appendClassName)("sui-facet", className)
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("div", {
    className: "sui-facet__title"
  }, label), /*#__PURE__*/_react.default.createElement("ul", {
    className: "sui-single-option-facet"
  }, value && /*#__PURE__*/_react.default.createElement("li", {
    className: "sui-single-option-facet__selected"
  }, (0, _viewHelpers.getFilterValueDisplay)(value), " ", /*#__PURE__*/_react.default.createElement("span", {
    className: "sui-single-option-facet__remove"
  }, "(", /*#__PURE__*/_react.default.createElement("a", {
    onClick: function onClick(e) {
      e.preventDefault();
      onRemove(value);
    },
    href: "/"
  }, "Remove"), ")")), !value && options.map(function (option) {
    return /*#__PURE__*/_react.default.createElement("li", {
      className: "sui-single-option-facet__item",
      key: (0, _viewHelpers.getFilterValueDisplay)(option.value)
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: "sui-single-option-facet__link",
      href: "/",
      onClick: function onClick(e) {
        e.preventDefault();
        onSelect(option.value);
      }
    }, (0, _viewHelpers.getFilterValueDisplay)(option.value)), " ", /*#__PURE__*/_react.default.createElement("span", {
      className: "sui-facet__count"
    }, option.count));
  }))));
}

SingleLinksFacet.propTypes = {
  label: _propTypes.default.string.isRequired,
  onRemove: _propTypes.default.func.isRequired,
  onSelect: _propTypes.default.func.isRequired,
  options: _propTypes.default.arrayOf(_types.FacetValue).isRequired,
  className: _propTypes.default.string
};
var _default = SingleLinksFacet;
exports.default = _default;
