// @elastic/react-search-ui-views/es/MultiCheckboxFacet.js
import PropTypes from "prop-types";
import React from "react";
import { FacetValue } from "./types";
import { appendClassName, getFilterValueDisplay } from "./view-helpers";

function MultiCheckboxFacet(_ref) {
  var className = _ref.className,
      label = _ref.label,
      onMoreClick = _ref.onMoreClick,
      onRemove = _ref.onRemove,
      onSelect = _ref.onSelect,
      options = _ref.options,
      showMore = _ref.showMore,
      showSearch = _ref.showSearch,
      onSearch = _ref.onSearch,
      searchPlaceholder = _ref.searchPlaceholder;
  return /*#__PURE__*/React.createElement("fieldset", {
    className: appendClassName("sui-facet", className)
  }, /*#__PURE__*/React.createElement("legend", {
    className: "sui-facet__title"
  }, label), showSearch && /*#__PURE__*/React.createElement("div", {
    className: "sui-facet-search"
  }, /*#__PURE__*/React.createElement("input", {
    className: "sui-facet-search__text-input",
    type: "search",
    placeholder: searchPlaceholder || "Search",
    onChange: function onChange(e) {
      onSearch(e.target.value);
    }
  })), /*#__PURE__*/React.createElement("div", {
    className: "sui-multi-checkbox-facet"
  }, options.length < 1 && /*#__PURE__*/React.createElement("div", null, "No matching options"), options.map(function (option) {
    var checked = option.selected;
    return /*#__PURE__*/React.createElement("label", {
      key: "".concat(getFilterValueDisplay(option.value)),
      htmlFor: "example_facet_".concat(label).concat(getFilterValueDisplay(option.value)),
      className: "sui-multi-checkbox-facet__option-label"
    }, /*#__PURE__*/React.createElement("div", {
      className: "sui-multi-checkbox-facet__option-input-wrapper"
    }, /*#__PURE__*/React.createElement("input", {
      id: "example_facet_".concat(label).concat(getFilterValueDisplay(option.value)),
      type: "checkbox",
      className: "sui-multi-checkbox-facet__checkbox",
      checked: checked,
      onChange: function onChange() {
        return checked ? onRemove(option.value) : onSelect(option.value);
      }
    }), /*#__PURE__*/React.createElement("span", {
      className: "sui-multi-checkbox-facet__input-text"
    }, getFilterValueDisplay(option.value))), /*#__PURE__*/React.createElement("span", {
      className: "sui-multi-checkbox-facet__option-count"
    }, option.count.toLocaleString("en")));
  })), showMore && /*#__PURE__*/React.createElement("button", {
    type: "button",
    className: "sui-facet-view-more",
    onClick: onMoreClick,
    "aria-label": "Show more options"
  }, "+ More"));
}

MultiCheckboxFacet.propTypes = {
  label: PropTypes.string.isRequired,
  onMoreClick: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(FacetValue).isRequired,
  showMore: PropTypes.bool.isRequired,
  className: PropTypes.string,
  showSearch: PropTypes.bool,
  searchPlaceholder: PropTypes.string
};
export default MultiCheckboxFacet;
