// @elastic/react-search-ui-views/es/SingleLinksFacet.js
import PropTypes from "prop-types";
import React from "react";
import { FacetValue } from "./types";
import { getFilterValueDisplay } from "./view-helpers";
import { appendClassName } from "./view-helpers";

function SingleLinksFacet(_ref) {
  var className = _ref.className,
      label = _ref.label,
      onRemove = _ref.onRemove,
      onSelect = _ref.onSelect,
      options = _ref.options;
  var value = options.filter(function (o) {
    return o.selected;
  }).map(function (o) {
    return o.value;
  })[0];
  return /*#__PURE__*/React.createElement("div", {
    className: appendClassName("sui-facet", className)
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "sui-facet__title"
  }, label), /*#__PURE__*/React.createElement("ul", {
    className: "sui-single-option-facet"
  }, value && /*#__PURE__*/React.createElement("li", {
    className: "sui-single-option-facet__selected"
  }, getFilterValueDisplay(value), " ", /*#__PURE__*/React.createElement("span", {
    className: "sui-single-option-facet__remove"
  }, "(", /*#__PURE__*/React.createElement("a", {
    onClick: function onClick(e) {
      e.preventDefault();
      onRemove(value);
    },
    href: "/"
  }, "Remove"), ")")), !value && options.map(function (option) {
    return /*#__PURE__*/React.createElement("li", {
      className: "sui-single-option-facet__item",
      key: getFilterValueDisplay(option.value)
    }, /*#__PURE__*/React.createElement("a", {
      className: "sui-single-option-facet__link",
      href: "/",
      onClick: function onClick(e) {
        e.preventDefault();
        onSelect(option.value);
      }
    }, getFilterValueDisplay(option.value)), " ", /*#__PURE__*/React.createElement("span", {
      className: "sui-facet__count"
    }, option.count));
  }))));
}

SingleLinksFacet.propTypes = {
  label: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(FacetValue).isRequired,
  className: PropTypes.string
};
export default SingleLinksFacet;
